app/controllers/admin/users_controller.rb

```diff
@@ -40,6 +40,30 @@ def update
       redirect_to "/admin/users/#{params[:id]}"
     end
 
+    def destroy
+      role = params[:role].to_sym
+      resource_type = params[:resource_type]
+
+      if role == :super_admin
+        flash[:danger] = "Super Admin roles cannot be removed."
+        redirect_to "/admin/users/#{params[:id]}/edit" and return
+      end
+
+      @user = User.find(params[:user_id])
+
+      if @user.id == current_user.id
+        flash[:danger] = "Admins cannot remove roles from themselves."
+      elsif role == :single_resource_admin
+        User.find(params[:user_id]).remove_role(role, resource_type)
+        flash[:success] = "Role: #{role.to_s.humanize.titlecase} has been successfully removed from the user!"
```

Comment:
I'm running into an issue where this feature is working a little _too_ well and rather than removing the particular `single_resource_admin` role with a click of the "X," it's removing _all_ `single_resource_admin` roles. 🙃 

That being said, within this `elsif` I'm trying to ensure that we _only_ remove the `single_resource_admin` role that is being clicked on to be removed. In order to remove a `single_resource_admin` role, we need to do the following:
```
User.remove_role(:single_resource_admin, RoleOfChoice)
```
My code is _almost_ working, in that it grabs the correct role, in almost the correct format, but where it is not working is that `resource_type` returns a string, but to properly remove a role, we need the instance of the resource rather than the string (as shown above in my code snippet). 

With all of this being said, I could use some help with the following:
1. Determining whether I'm approaching the removal of `single_resource_admin` roles correctly.
2. If it has been determined that I am approaching the removal of `single_resource_admin` roles correctly, then being pointed in the right direction for how I might adjust my code to return the `resource_type` in the necessary format, rather than as a string.
3. If it has been determined that I am not approaching the removal of `single_resource_admin` roles correctly, then being pointed in the right direction for how I might approach the removal of `single_resource_admin` roles differently.

I have looked at the Rolify docs, Forem docs, dev.to, SO, and other random QA sites for possible solutions for insight on removing a resource instance and have been unsuccessful in my searches. Additionally, I've tried removing the specific `single_resource_admin` role via `resource_name`, `resource_type`, and `resource_id` in various ways both in the `Admin::UsersController` and in the `_current_roles.html.erb` partial, with the help of my tests and `binding.pry`. Unfortunately, none of these attempts have gotten me quite where I need to be.

Any and all help is welcome and greatly appreciated at this point! 

cc @citizen428

Reply:
@juliannatetreault I had written this long answer but forgot to publish the review 🙄 I'm unresolving this to make sure you see it.

Reply:
Thank you so much for taking a look and helping me out with this @citizen428! Everything is working as expected now. 🎉  I used `.safe_constantize` instead of `.constantize` since I read that it was better for security reasons. Code Climate doesn't seem happy with either of the two methods, though. 

In regard to extracting this out into a service: do you think that would warrant a follow-up PR or should it be done in this PR?

Reply:
@juliannatetreault Good call on `safe_constantize` vs `constantize`. To be fair, the way we'd be using `constantize` here is not the pattern that leads to security issues (which is more along the lines of `something.constantize.new.method_call`), but it's a good habit to get into.

Re the service object, I'll leave it up to you. If you think it's beneficial to get this out faster, let's just deploy it and you do a follow-up.

Reply:
Update: circling back around to extracting the `#destroy` action into a service per your suggestion @citizen428!